src/System.Private.CoreLib/shared/System/Threading/ThreadPool.cs

```diff
@@ -951,32 +1005,43 @@ public static partial class ThreadPool
             WaitOrTimerCallback callBack,
             object state,
             long millisecondsTimeOutInterval,
-            bool executeOnlyOnce)
+            bool executeOnlyOnce    // NOTE: we do not allow other options that allow the callback to be queued as an APC
+        )
         {
-            if (millisecondsTimeOutInterval < -1 || millisecondsTimeOutInterval > int.MaxValue)
```

Comment:
This is checking that casting long to int won't drop upper bits. Looks like we are losing a bug fix made in CoreRT here.

Reply:
(ok to do open issue on this and deal with it separately)

Reply:
I will have follow up PR moving more ThreadPool code to shared. Will add it there.